schemas/com.snowplowanalytics.snowplow/application_error/jsonschema/1-0-2

```diff
@@ -0,0 +1,90 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Schema for an application error",
+  "self": {
+    "vendor": "com.snowplowanalytics.snowplow",
+    "name": "application_error",
+    "format": "jsonschema",
+    "version": "1-0-2"
+  },
+
+  "type": "object",
+  "properties": {
+    "programmingLanguage": {
+      "enum": [
+        "JAVA",
+        "SCALA",
+        "KOTLIN",
+        "GROOVY",
+        "RUBY",
+        "GOLANG",
+        "JAVASCRIPT",
+        "PHP",
+        "PYTHON",
+        "OBJECTIVEC",
+        "SWIFT",
+        "C",
+        "CPLUSPLUS",
+        "CSHARP",
+        "ACTIONSCRIPT",
+        "LUA",
+        "RUST",
+        "HASKELL",
+        "CLOJURE",
+        "ERLANG",
+        "ELIXIR",
+        "CRYSTAL",
+        "PONY",
+        "NIM"
+      ]
+    },
+    "message": {
+      "type": "string",
+      "maxLength": 2048
+    },
+    "threadName": {
+      "type": ["string", "null"],
+      "maxLength": 1024
+    },
+    "threadId": {
+      "type": ["integer", "null"],
+      "minimum": 0,
+      "maximum": 2147483647
+    },
+    "stackTrace": {
+      "type": ["string", "null"],
+      "maxLength": 8192
+    },
+    "causeStackTrace": {
+      "type": ["string", "null"],
+      "maxLength": 8192
```

Comment:
are we sure about the length of stack traces?

Reply:
Actually, not at all, I don't even think there's an upper bound. But I guess good news is that usually most useful information is on top of stacktrace and it is intended to be human-readable, so we don't care that much if it was truncated.

Reply:
yes that's what I meant, will I have to truncate my stack traces to integrate with this schema and do I want to be doing that?

Reply:
Yes, we'll have to truncate it. Though these events are usually sent via middleware, such as trackers or processing manifest and it is up to middleware to construct valid payload. We already truncate it in Scala tracker and I think must do it in all other trackers.

Reply:
...and obviously in processing manifest as well.